### PR TITLE
Feat/methods using facets

### LINKED ISSUE #49: Complete Surface and Solid API using edge/facet indices

### PR DESCRIPTION
**Pull request by panquez**
fix #49

### CODE REVIEW

File: include/geode/mesh/core/detail/facet_storage.h [hunk 1]
```diff
@@ -44,7 +44,11 @@ namespace geode
                 : counter_(
                     facet_attribute_manager_
                         .template find_or_create_attribute< VariableAttribute,
-                            index_t >( "counter", 1 ) )
+                            index_t >( "counter", 1 ) ),
+                  vertices_(
+                      facet_attribute_manager_
+                          .template find_or_create_attribute< VariableAttribute,
+                              std::vector< index_t > >( "edge_vertices" ) )
```
Comment (BotellaA) [inline include/geode/mesh/core/detail/facet_storage.h]: Always vector ? Not array for edges ?

File: include/geode/mesh/core/detail/facet_storage.h [hunk 2]
```diff
@@ -150,9 +156,15 @@ namespace geode
                     }
                     VertexCycle updated_cycle{ updated_vertices };
                     facet_indices_[updated_cycle] = cycle.second;
+                    vertices_->value( cycle.second ) = updated_vertices;
                 }
             }
 
+            std::vector< index_t > get_facet_vertices( index_t facet_id ) const
```
Comment (BotellaA) [inline include/geode/mesh/core/detail/facet_storage.h]: Is the copy necessary ?

Comment (panquez) [reply include/geode/mesh/core/detail/facet_storage.h]: No, removed

File: src/geode/mesh/core/polygonal_surface.cpp [hunk 3]
```diff
@@ -138,7 +141,7 @@ namespace geode
 
         std::array< index_t, 2 > get_edge_vertices( index_t edge_id ) const
         {
-            auto vertices = this->get_facet_vertices( edge_id );
+            const auto& vertices = this->get_facet_vertices( edge_id );
```
Comment (BotellaA) [inline src/geode/mesh/core/polygonal_surface.cpp]: directly return it with cont&

File: upgrade-guide.md [hunk 4]
```diff
@@ -9,6 +9,26 @@ TODO
...
+
+```
+PolygonEdge polygon_edge{ 0, 0 };
+auto edge_length = surface.polygon_edge_length( polygon_edge );
+```
+
+to 
+
+```
+index_t edge_id = polygon_edge( { 0, 0 } );
```
Comment (BotellaA) [inline upgrade-guide.md]: it could be auto

Comment (panquez) [reply upgrade-guide.md]: I agree with you but I explicitly write index_t to show that edge_length() takes an index_t. What is your opinion?

Comment (BotellaA) [reply upgrade-guide.md]: ok

**Pull request merged**